src/Stratis.Features.FederatedPeg/Wallet/MultiSigTransactions.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+
+namespace Stratis.Features.FederatedPeg.Wallet
+{
+    public class MultiSigTransactions : ICollection<TransactionData>, ITransactionDataObserver
+    {
+        private readonly object lockObject;
+        private Dictionary<OutPoint, TransactionData> transactionDict;
+        private Dictionary<OutPoint, TransactionData> spendableTransactionDict;
+        private Dictionary<uint256, List<TransactionData>> withdrawalsByDepositDict;
+        private SortedDictionary<int, List<TransactionData>> spentTransactionsByHeightDict;
+
+        public int Count => this.transactionDict.Count;
+
+        public bool IsReadOnly => false;
```

Comment:
What is is for?

Reply:
It's required by the ICollection interface.